src/algebra/star/self_adjoint.lean

```diff
@@ -37,15 +37,117 @@ We also define `is_star_normal R`, a `Prop` that states that an element `x` sati
 
 -/
 
-variables (R : Type*) {A : Type*}
+variables {R A : Type*}
+
+/-- An element of a star monoid is self-adjoint if it is equal to its adjoint. -/
+def is_self_adjoint [has_star R] (x : R) : Prop := star x = x
+
+/-- An element of a star monoid is normal if it commutes with its adjoint. -/
+class is_star_normal [has_mul R] [has_star R] (x : R) : Prop :=
+(star_comm_self : commute (star x) x)
+
+export is_star_normal (star_comm_self)
+
+lemma star_comm_self' [has_mul R] [has_star R] (x : R) [is_star_normal x] :
+  (star x) * x = x * star x :=
+is_star_normal.star_comm_self
+
+namespace is_self_adjoint
+
+lemma star_eq [has_star R] {x : R} (hx : is_self_adjoint x) : star x = x := hx
+
+lemma _root_.is_self_adjoint_iff [has_star R] {x : R} : is_self_adjoint x ↔ star x = x := iff.rfl
+
+section add_group
+variables [add_group R] [star_add_monoid R]
+
+variables (R)
+
+lemma _root_.is_self_adjoint_zero : is_self_adjoint (0 : R) := star_zero R
+
+variables {R}
+
+lemma add {x y : R} (hx : is_self_adjoint x) (hy : is_self_adjoint y) : is_self_adjoint (x + y) :=
+by { rw is_self_adjoint_iff at ⊢ hx hy, simp only [hx, hy, star_add] }
```

Comment:
I am thinking of rewriting all of these statements using `star_eq`, then it would be `by simp only [..]`. While your suggestion is more streamlined, it makes all the proofs look very different even though they are in some sense all the same (unfold the definition and use elementary facts about the star).

Reply:
@hrmacbeth what do you think of the current version, i.e.,
`by simp only [is_self_adjoint_iff, star_add, hx.star_eq, hy.star_eq]` ?